hack/make-rules/test-cmd-util.sh

```diff
@@ -3117,6 +3128,17 @@ run_daemonset_history_tests() {
   # Clean up
   kubectl delete -f hack/testdata/rollingupdate-daemonset.yaml "${kube_flags[@]}"
 
+  ### Set image of a DaemonSet 
```

Comment:
It's a little weird to put this test under `run_daemonset_history_tests`, since there is no history involved in this test case. 
I'd suggest put both new test cases under a new run_kubectl_set_image_tests.

Reply:
I meant to put it in `run_daemonset_tests`. I'd prefer put it in rs test and ds test, instead of kubectl set image test, because it's more resource specific (we're not testing functionality of kubectl set image, but how it works with different resources). WDYT?

Reply:
To be more clear
```
run_kubectl_set_image_tests() {
  # some test for kubectl set RS

  # some test for kubectl set daemonset
}
```